README.md

```diff
@@ -10,33 +10,115 @@ This repository contains a Terraform provider for the [Materialize platform](htt
 * [Terraform](https://www.terraform.io/downloads.html) >= 1.0.3
 * [Go](https://golang.org/doc/install) >= 1.16
 
-## Developing the provider
+## Installation
 
-If you wish to work on the provider, you'll first need
-[Go](http://www.golang.org) installed on your machine (see
-[Requirements](#requirements) above).
+To use the provider, add the following configuration to your Terraform settings:
 
-To compile the provider, run `make install`. This will build the provider and put the provider binary in the correct location within `~/.terraform.d` so that Terraform can find the plugin.
+```hcl
+terraform {
+  required_providers {
+    materialize = {
+      source = "materialize.com/devex/materialize"
+    }
+  }
+}
+```
+
+Configure the provider by adding the following block to your Terraform project:
+
+```hcl
+provider "materialize" {
+  host     = "materialized_hostname"
+  username = "materialize_username"
+  password = "materialize_password"
+  port     = 6875
+  database = "materialize"
+}
+```
+
+Once you have configured the provider, you can start defining resources using Terraform. You can find examples of how to define resources in the [`examples`](./examples/) directory.
+
+## Usage
+
+### Managing resources
+
+You can manage resources using the `terraform apply` command. For example, to create a new connection named `kafka_connection`, add the following resource definition to your Terraform project:
+
+```hcl
+# main.tf
+resource "materialize_connection_kafka" "kafka_connection" {
+  name = "kafka_connection"
+  kafka_broker {
+    broker = "b-1.hostname-1:9096"
+  }
+  sasl_username = "example"
+  sasl_password {
+    name          = "kafka_password"
+    database_name = "materialize"
+    schema_name   = "public"
+  }
+  sasl_mechanisms = "SCRAM-SHA-256"
+  progress_topic  = "example"
+}
+```
+
+Then, run apply the changes:
+
+```bash
+terraform apply
+```
+
+### Data sources
 
-To generate or update documentation, run `go generate`.
+You can use data sources to retrieve information about existing resources. For example, to retrieve information about the existing sinks in your Materialize instance, add the following data source definition to your Terraform project:
 
-To run the full suite of acceptance tests, run `make testacc`.
+```hcl
+# main.tf
+data "materialize_connection" "all" {}
 
-```sh
-$ make testacc
+output name {
+  value       = data.materialize_connection.all
+}
 ```
 
-### Adding dependencies
+Then, check the Terraform plan:
 
-This provider uses [Go modules](https://github.com/golang/go/wiki/Modules). Please see the Go documentation for the most up to date information about using Go modules.
+```bash
+terraform plan
+```
+
+### Importing existing resources
+
+You can import existing resources into your Terraform state using the `terraform import` command. For example, to import an existing connection named `kafka_connection`, first add the resource definition to your Terraform project:
```

Comment:
Worth mentioning explicitly that ids are on the ids from the mz catalogue and link that https://materialize.com/docs/sql/system-catalog/mz_catalog/